File: problem/Min_Max_Pair_Sum/minPairSum.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minPairSum = function(nums) {
    nums.sort((a, b) => a - b);
    let res = 0;
    let n = nums.length;
    
    for (let i = 0; i < n >> 1; i++) {
        res = Math.max(res, nums[i] + nums[n - 1 - i]);
    }
    
    return res;
};

File: problem/Min_Max_Pair_Sum/minPairSum.c
int cmp(const void *a, const void *b)
{
    return (*(int *)a - *(int *)b);
}

int minPairSum(int *nums, int numsSize)
{
    qsort(nums, numsSize, sizeof(int), cmp);
    int res = 0;

    for (int i = 0; i < numsSize / 2; i++)
    {
        int current = nums[i] + nums[numsSize - 1 - i];
        if (current > res)
        {
            res = current;
        }
    }

    return res;
}